### DISCUSSION TITLE
Improve response time with load balancer

### DISCUSSION THREAD

**Discussion by namhq1989** [minio/minio#15150] [Q&A] [answered]
I have 4 instances of Minio, and 1 load balancer (HAProxy). It is currently stable , but I have a problem with response time. If I load an url with HAProxy, response time will be much longer than load directly from Minio instance. Example:

- load from http://103.153.74.137:9000/files/md_2022_06_17_08_07_09_0700_cSGBJBayly.jpg, response time is 100-200ms
- load from https://media.sellycdn.net/files/md_2022_06_17_08_07_09_0700_cSGBJBayly.jpg, response time is 3-5s

-------
- HAProxy config
```
frontend app
  bind :80
  bind :443 ssl crt /etc/haproxy/certs/sellycdn.net.pem
  http-request redirect scheme https unless { ssl_fc }
  option forwardfor

  # media.sellycdn.net
  use_backend minioserver if { hdr(host) -i media.sellycdn.net }

backend minioserver
  server minio1 minio-1:9000 check
  server minio2 minio-2:9000 check
  server minio3 minio-3:9000 check
  server minio4 minio-4:9000 check
  balance roundrobin
```

Can someone tell me why and how to improve response time with load balancer?

**Comment by harshavardhana** [comment, marked as answer]
This is not MinIO's problem - you should ask haproxy folks.
